janus_server/src/config.rs

```diff
@@ -0,0 +1,110 @@
+//! Configuration for various Janus actors
+
+use crate::message::{Duration, HpkeConfig, Role, TaskId};
+use serde::{Deserialize, Serialize};
+use std::net::SocketAddr;
+use url::Url;
+
+/// Configuration for a Janus server using a database.
+#[derive(Clone, Debug, PartialEq, Eq, Serialize, Deserialize)]
+pub struct DbConfig {
+    /// URL at which to connect to the database.
+    pub url: Url,
+    // TODO: add option for connecting to database over TLS, if necessary
+}
+
+/// Non-secret configuration options for a Janus aggregator, deserialized from
+/// YAML.
+///
+/// # Examples
+///
+/// ```
+/// use janus_server::config::AggregatorConfig;
+///
+/// let yaml_config = r#"
+/// ---
+/// listen_address: "0.0.0.0:8080"
+/// database:
+///   url: "postgres://postgres:postgres@localhost:5432/postgres"
+/// task_id: 41524c07214502416f56b1258d97521b083483a0fbb082464e0741ace1403a2e
+/// role: Leader
+/// tolerable_clock_skew_seconds: 10
+/// hpke_config:
+///   id: 0
+///   kem_id: X25519HkdfSha256
+///   kdf_id: HkdfSha512
+///   aead_id: ChaCha20Poly1305
+///   public_key: a6f620ccc9d35164d92a4073727645f20329e1e1fe6bac7400c75a627c64783d
+/// "#;
+///
+/// let _decoded: AggregatorConfig = serde_yaml::from_str(yaml_config).unwrap();
+/// ```
+#[derive(Clone, Debug, PartialEq, Eq, Serialize, Deserialize)]
+pub struct AggregatorConfig {
+    /// Address on which this server should listen for connections and serve its
+    /// API endpoints.
+    // TODO: Options for terminating TLS, unless that gets handled in a load
+    // balancer?
+    pub listen_address: SocketAddr,
+    /// The aggregator's database configuration.
+    pub database: DbConfig,
+    /// The PPM task this aggregator is participating in. Parameters for this
+    /// task must exist in the aggregator's database.
+    // TODO: if we want a single aggregator instance to participate in multiple
+    // tasks, then we should provide a list or map of per-task parameters
+    pub task_id: TaskId,
+    /// The PPM [`Role`] this aggregator plays. Should be either
+    /// [`Role::Leader`] or [`Role::Helper`].
+    pub role: Role,
```

Comment:
Following the above discussion about `task_id`, I think `role` should not be in the config either since it is a piece of task-specific configuration that should instead live in the DB.